modules/mobile-agents/android/client/src/org/wso2/mdm/agent/AlertActivity.java

```diff
@@ -63,4 +89,46 @@ public boolean onCreateOptionsMenu(com.actionbarsherlock.view.Menu menu) {
 		return true;
 	}
 
+	/**
+	 * This method stops the device ring.
+	 */
+	private void stopRing() {
+
+		if (defaultRingtone != null && defaultRingtone.isPlaying()) {
+			defaultRingtone.stop();
+		}
+	}
+
+	/**
+	 * This method is used to start ringing the phone.
+	 */
+	@TargetApi(21)
+	private void startRing() {
+		defaultRingtoneUri = RingtoneManager.getActualDefaultRingtoneUri(this, RingtoneManager.TYPE_RINGTONE);
+		defaultRingtone = RingtoneManager.getRingtone(this, defaultRingtoneUri);
+
+		if (deviceInfo.getSdkVersion() >= Build.VERSION_CODES.LOLLIPOP) {
+			AudioAttributes attributes = new AudioAttributes.Builder().
+					setUsage(AudioAttributes.USAGE_NOTIFICATION).
+					setContentType(AudioAttributes.CONTENT_TYPE_SONIFICATION).
+					build();
+			defaultRingtone.setAudioAttributes(attributes);
+			defaultRingtone.play();
```

Comment:
defaultRingtone.play() can be brought outside the if blocks and be called after as it in anyway executes finally in both blocks.